lib/Service/Gateway/SMS/Provider/AndroidGSMmodem.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OCA\TwoFactorGateway\Service\Gateway\SMS\Provider;
+
+use Exception;
+use OCA\TwoFactorGateway\Exception\SmsTransmissionException;
+use OCP\Http\Client\IClient;
+use OCP\Http\Client\IClientService;
+
+class AndroidGsmModem implements IProvider {
+
+	const PROVIDER_ID = 'android_gsm_modem';
+
+	/** @var IClient */
+	private $client;
+
+	/** @var AndroidGsmModemConfig */
+	private $config;
+
+	public function __construct(IClientService $clientService,
+								AndroidGsmModemConfig $config) {
+		$this->client = $clientService->newClient();
+		$this->config = $config;
+	}
+
+	/**
+	 * @param string $identifier
+	 * @param string $message
+	 *
+	 * @throws SmsTransmissionException
+	 */
+	public function send(string $identifier, string $message) {
+		$config = $this->getConfig();
+		$user = $config->getUser();
+		$password = $config->getPassword();
+		$host = $config->getHost();
+		try {
+			$this->client->get("http://$host/SendSMS", [
```

Comment:
what if anyone secures the connection with SSL? how about moving the scheme part to `$host` and letting admins specify in the config?

Reply:
As far as I'm aware the app that is installed on the phone itself only supports http connections, if https connections were supported I'd add a config for it 👍

Reply:
Although actually if it was served via a secured reverse proxy, then this would require a config, I'll look into it.

Reply:
Yep, that is the scenario I was thinking about. Thanks :v: